### DISCUSSION TITLE
Managing Dependency

### DISCUSSION THREAD

**Discussion by ritab-snow** [fal-ai/dbt-fal#414] [Q&A] [answered]
When we look at the execution order of dbt models, the python script always runs at the end of all the models.
Order of model execution can be handled using {{ref()}}. Is there a way that we can control the order of execution for the python script as well ? I want to run my models in the below order. 
```
├── models
│   ├── aggregate
│   │   └── aggr_distance_travelled.sql.  [SEQ5]
│   ├── staging
│   │   ├── base_view_trips.sql.     [SEQ1]
│   │   ├── schema.yml
│   │   └── sources.yml
│   └── transform
│       ├── tfm_trip_aggr.sql	     [SEQ3]
│       └── tfm_trip_distance.sql.   [SEQ2]
├── python_transform
│   └── tfm_getzipcode.py            [SEQ4]
```

How can I make this possible ?

**Comment by chamini2** [comment, marked as answer]
I guess this probably means `tfm_getzipcode` write to the data warehouse? In that case, I would recommend you move to [Python Data Models](https://docs.fal.ai/Docs/fal-cli/python-models), a new way of writing fal scripts to be represented as a dbt model.

**Reply by ritab-snow** [reply]
Thanks you for your response. I am aware that we can write back to the data warehouse from python scripts. It's a wonderful feature. In my project I wanted to perform some heavy weight aggregation using a .sql file in the data warehouse itself after the python script is run. If this is limitation now then I will have to add the aggregation step within the python script as well.
Can I conclude that as of this version of fal, the python script will always be the last step in the workflow ?

**Reply by chamini2** [reply]
No, Python _can_ be in the middle of your workflow. You can achieve that with [Python Data Models](https://docs.fal.ai/Docs/fal-cli/python-models). Can you check out those docs and see if it fits your setup?

You can also reach out in [Discord](https://discord.com/invite/Fyc9PwrccF) for faster feedback if you want.

**Reply by ritab-snow** [reply]
Thanks again. I checked the document but then I did not find an answer if a python script can be run somewhere in the middle of a dbt model. If you can share some quick links that will really help.

**Comment by ritab-snow** [comment]
Many Thanks. I tried this. It works !!

Regards,
Ritab
